src/core/scheduler_utils.cc

```diff
@@ -270,11 +270,18 @@ Scheduler::Payload
 PriorityQueue::Dequeue()
 {
   pending_cursor_.valid_ = false;
-  if (!queues_[front_priority_level_].Empty()) {
-    size_--;
-    return queues_[front_priority_level_].Dequeue();
-  } else if (front_priority_level_ != last_priority_level_) {
-    front_priority_level_++;
+  while (true) {
+    if (!queues_[front_priority_level_].Empty()) {
+      size_--;
+      return queues_[front_priority_level_].Dequeue();
+    } else if (front_priority_level_ != last_priority_level_) {
+      front_priority_level_++;
+      continue;
+    }
+
+    // Control reach here if the queue for last priority level is also empty,
+    // then raise exception
+    break;
   }
   throw std::out_of_range("dequeue on empty queue");
```

Comment:
I just put throw here as the control shouldn't reach here, as the batcher should only dequeue on non-empty queue. Otherwise, that means the book keeping goes wrong.

Yes we can log it and recover from the error, but that somewhat hides the actual cause.

Reply:
Since we can recover it is better to log an error and continue... we don't want to crash the server.

Reply:
Ok, will change it.